Release/src/utilities/asyncrt_utils.cpp

```diff
@@ -736,6 +736,12 @@ static compute_year_result compute_year(int64_t secondsSince1900)
     int secondsInt = static_cast<int>(secondsLeft - year4 * SecondsIn4Years);
 
     int year1 = secondsInt / SecondsInYear;
+    if (year1 == 4)
+    {
+        // this is the last day in a leap year
+        year1 = 3;
+    }
```

Comment:
Given this is being ported here, I want to understand it a bit better:
https://github.com/Azure/azure-sdk-for-cpp/pull/1254

> However, it has an edge case where the current year is the last day of a leap year, such that `SecondsIn4Years` contains that extra leap day but the calculation of `year` didn't account for the current date being exactly that extra day.

I don't fully follow what you mean here. Can you please elaborate, with a concrete example. Also, what do you mean by the "current **year** is the last **day** of a leap year"?

Lastly, what is an example impact of this bug exactly to an end user?

cc @antkmsft

Reply:
> I don't fully follow what you mean here. Can you please elaborate, with a concrete example.

2020-12-31 results in year1 being 4, despite this code expecting all blocks of 4 years to be handled by SecondsIn4Years.

> Also, what do you mean by the "current year is the last day of a leap year"?

Should be current "current **day** is the last day of a leap year"

>Lastly, what is an example impact of this bug exactly to an end user?

Reply:
>Lastly, what is an example impact of this bug exactly to an end user?

It incorrectly prints Thursday 2020-12-31 as Thu 2021-01-01